webui/react/src/components/TrialActions.tsx

```diff
@@ -0,0 +1,72 @@
+import { Button } from 'antd';
+import React, { useCallback, useState } from 'react';
+
+import { ButtonLoadingStates, ConditionalButton } from 'components/ExperimentActions';
```

Comment:
**suggestion:** It seems odd to pull in shared items from a sibling component. Take a look at the composition setup for `SelectFilter`, `UserSelectFilter` and `StateSelectFilter`. Similar to an inheritance but React's version of it.

Reply:
I think this case is a bit simpler we really just need to share a single type definition. how about a types.ts file under `components/` to share component and react specific types?

Reply:
let's keep it as how you have it for now, just because there are some ideas I think we can implement to improve this situation (without composition).

Reply:
nm, just saw your changes to `component/types` that's good too